streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/ThreadMetrics.java

```diff
@@ -146,18 +151,46 @@ public static Sensor pollSensor(final String threadId,
         );
     }
 
-    public static Sensor processSensor(final String threadId,
-                                       final StreamsMetricsImpl streamsMetrics) {
-        return invocationRateAndCountAndAvgAndMaxLatencySensor(
-            threadId,
-            PROCESS,
-            PROCESS_RATE_DESCRIPTION,
-            PROCESS_TOTAL_DESCRIPTION,
+    public static Sensor processLatencySensor(final String threadId,
+                                              final StreamsMetricsImpl streamsMetrics) {
+        final Sensor sensor = streamsMetrics.threadLevelSensor(threadId, PROCESS, RecordingLevel.INFO);
+        final Map<String, String> tagMap = streamsMetrics.threadLevelTagMap(threadId);
+        final String threadLevelGroup = threadLevelGroup(streamsMetrics);
+        addAvgAndMaxToSensor(
+            sensor,
+            threadLevelGroup,
+            tagMap,
+            PROCESS + LATENCY_SUFFIX,
             PROCESS_AVG_LATENCY_DESCRIPTION,
-            PROCESS_MAX_LATENCY_DESCRIPTION,
-            Sensor.RecordingLevel.INFO,
-            streamsMetrics
+            PROCESS_MAX_LATENCY_DESCRIPTION
+        );
+        return sensor;
+    }
+
+    public static Sensor processRateSensor(final String threadId,
+                                              final StreamsMetricsImpl streamsMetrics) {
+        final Sensor sensor = streamsMetrics.threadLevelSensor(threadId, PROCESS, RecordingLevel.INFO);
+        final Map<String, String> tagMap = streamsMetrics.threadLevelTagMap(threadId);
+        final String threadLevelGroup = threadLevelGroup(streamsMetrics);
+        sensor.add(
```

Comment:
We can still leverage on `addInvocationRateToSensor` here?

Reply:
No, because we're specifically not measuring the rate of `sensor.record(quantity)` invocations, which means to count each `record` event as `1`, regardless of `quantity`. Instead, we actually record the `quantity` now.

That sounds clear as mud to me, so here's an example:

```
Sensor S
* Metric A := invocation rate sensor
* Metric B := "plain" rate sensor

Execute: S.record(2) once in a second
State:
A.rate == 1/sec
A.total == 1
B.rate == 2/sec
B.total == 2
```

In this example, `B` is the definition we are aiming for.

Reply:
Yeah I know that :) I was referring to the one with `SUM` not `COUNT` too but I copied the wrong name. We already have a public `addRateOfSumMetricToSensor` which is only used for rocksDB today, and I was wondering if we can leverage that.

Reply:
Ah, sorry :)

I saw that metric, but had no idea what a "rate of sum" is... I fixed it now.